File: tests/fixtures/vulnerable/insecure_cookies.py
"""Vulnerable: insecure cookie settings."""
from flask import make_response


def set_session_cookie(response, token):
    """Bad: SameSite=None without Secure, no HttpOnly."""
    response.set_cookie(
        "session",
        token,
        samesite="None",
    )
    return response


def set_refresh_token(response, refresh):
    """Bad: no HttpOnly on refresh token cookie."""
    response.set_cookie(
        "refresh_token",
        refresh,
        domain=".example.com",
    )
    return response

File: tests/fixtures/vulnerable/insecure_cookies.ts
// Vulnerable: insecure cookie settings
import express from "express";

const app = express();

app.get("/set-session", (req, res) => {
  // Bad: SameSite=None without secure, no httpOnly
  res.cookie("session", "token123", {
    sameSite: "none",
  });
  res.json({ status: "ok" });
});

app.get("/set-refresh", (req, res) => {
  // Bad: no httpOnly on refresh token
  res.cookie("refresh_token", "refresh123", {
    domain: ".example.com",
  });
  res.json({ status: "ok" });
});
